test/byContract/registries/registries.test.ts

```diff
@@ -0,0 +1,398 @@
+            anyone,
+        );
+        datasetRegistryAsAdmin = datasetRegistry.connect(iexecAdmin);
+        workerpoolRegistry = WorkerpoolRegistry__factory.connect(
+            await iexecPoco.workerpoolregistry(),
+            anyone,
+        );
+        workerpoolRegistryAsAdmin = workerpoolRegistry.connect(iexecAdmin);
+    }
+
+    describe('initialize', () => {
+        it('Should initialize new deployed registries', async () => {
+            const newAppRegistry = await new AppRegistry__factory()
+                .connect(iexecAdmin)
+                .deploy()
+                .then((contract) => contract.deployed());
+            await newAppRegistry.initialize(appRegistry.address);
+            expect(await newAppRegistry.initialized()).to.be.true;
+            expect(await newAppRegistry.previous()).to.equal(appRegistry.address);
+
+            const newDatasetRegistry = await new DatasetRegistry__factory()
+                .connect(iexecAdmin)
+                .deploy()
+                .then((contract) => contract.deployed());
+            await newDatasetRegistry.initialize(datasetRegistry.address);
+            expect(await newDatasetRegistry.initialized()).to.be.true;
+            expect(await newDatasetRegistry.previous()).to.equal(datasetRegistry.address);
+
+            const newWorkerpoolRegistry = await new WorkerpoolRegistry__factory()
+                .connect(iexecAdmin)
+                .deploy()
+                .then((contract) => contract.deployed());
+            await newWorkerpoolRegistry.initialize(workerpoolRegistry.address);
+            expect(await newWorkerpoolRegistry.initialized()).to.be.true;
+            expect(await newWorkerpoolRegistry.previous()).to.equal(workerpoolRegistry.address);
+        });
+        it('Should not call initialize when user is not the owner', async () => {
+            await expect(appRegistry.initialize(AddressZero)).to.be.revertedWith(
+                'Ownable: caller is not the owner',
+            );
+            await expect(datasetRegistry.initialize(AddressZero)).to.be.revertedWith(
+                'Ownable: caller is not the owner',
+            );
+            await expect(workerpoolRegistry.initialize(AddressZero)).to.be.revertedWith(
+                'Ownable: caller is not the owner',
+            );
+        });
+        it('Should not reinitialize', async () => {
+            await expect(appRegistryAsAdmin.initialize(AddressZero)).to.be.revertedWithoutReason();
+            await expect(
+                datasetRegistryAsAdmin.initialize(AddressZero),
+            ).to.be.revertedWithoutReason();
+            await expect(
+                workerpoolRegistryAsAdmin.initialize(AddressZero),
+            ).to.be.revertedWithoutReason();
+        });
+    });
+
+    describe('setName', () => {
+        it('should set the ENS name for registries', async () => {
+            const appRegistryENSName = 'myAppRegistry.eth';
```

Comment:
Minor, you can update all you `somethingENSVar` to `somethingEnsVar` to follow practice we use so far